File: tests/fixtures/openai-realtime/openai-realtime-buffer-audio-until-session-ready-fixed/02-adversarial-different-readystate-check.ts
import WebSocket from 'ws';

// Adversarial: this checks readyState against CLOSED, not OPEN, so the
// "drop audio while not yet open" pattern this rule targets doesn't apply.
function send(socket: WebSocket, message: object, logger: any) {
  if (socket.readyState !== WebSocket.CLOSED) {
    socket.send(JSON.stringify(message));
  } else {
    logger.error('Socket permanently closed.');
  }
}

export default send;

File: tests/fixtures/openai-realtime/openai-realtime-buffer-audio-until-session-ready-fixed/02_adversarial_different_readystate_check.py
from websockets.protocol import State

# Adversarial: a readyState check against CLOSED, not OPEN — out of scope for
# this rule (it targets the OPEN-gated buffering pattern specifically).


async def send_audio_chunk(ws, chunk):
    if ws.state == State.CLOSED:
        logger.warning("socket closed, dropping chunk")
        return
    await ws.send(chunk)
